### PR TITLE
Bug 1860393:  LogForwarding GA on 4.5

### PR DESCRIPTION
**Pull request by jcantrill**
This PR brings LogForwarding GA into 4.5

Ref: 
* https://issues.redhat.com/browse/LOG-829
* https://bugzilla.redhat.com/show_bug.cgi?id=1860393

### CODE REVIEW

File: pkg/generators/forwarding/fluentd/output_conf_syslog_test.go [hunk 1]
```diff
@@ -142,23 +142,24 @@ var _ = Describe("Generating external syslog server output store config blocks",
         severity debug
     	protocol tcp
     	packet_size 4096
-		timeout 60
-		timeout_exception true
-	    keep_alive true
+	timeout 60
```
Comment (alanconway) [inline pkg/generators/forwarding/fluentd/output_conf_syslog_test.go:145]: Inconsistent indents, did this get a `go fmt`?

File: pkg/generators/forwarding/fluentd/output_conf_syslog_test.go [hunk 2]
```diff
@@ -142,23 +142,24 @@ var _ = Describe("Generating external syslog server output store config blocks",
...
+    	  @type file
+    	  path '/var/lib/fluentd/syslog_receiver'
+    	  flush_interval "#{ENV['ES_FLUSH_INTERVAL'] || '1s'}"
+    	  flush_thread_count "#{ENV['ES_FLUSH_THREAD_COUNT'] || 2}"
+    	  flush_at_shutdown "#{ENV['FLUSH_AT_SHUTDOWN'] || 'false'}"
+    	  retry_max_interval "#{ENV['ES_RETRY_WAIT'] || '300'}"
+    	  retry_forever true
+    	  queue_limit_length "#{ENV['BUFFER_QUEUE_LIMIT'] || '32' }"
+    	  chunk_limit_size "#{ENV['BUFFER_SIZE_LIMIT'] || '8m' }"
+       	  total_limit_size "#{ENV['TOTAL_LIMIT_SIZE'] ||  8589934592 }" #8G
```
Comment (alanconway) [inline pkg/generators/forwarding/fluentd/output_conf_syslog_test.go:161]: Can this be written 8g for fluent?

**Comment by markmc** [comment]
@jcantrill note this PR is on hold, and needs a rebase

**Comment by jcantrill** [comment]
Closing WONTFIX as the decision was made to not backport to 4.5.z given workarounds exist to satisfy customer requirements

**Pull request closed without merging**